app/public/Model.js

```diff
@@ -49,6 +57,22 @@ export default class Model extends Observable {
 
         this.loginEndpoint = `/api${RCT.endpoints.login}`;
         this.login('physicist');
+
+        this.handleLocationChange();
+    }
+
+    /**
+     * Delegates sub-model actions depending on new location of the page
+     * @returns {void}
+     */
+    async handleLocationChange() {
```

Comment:
This function is meant to handle all the views once they are implemented in a similar way to periods.